src/main/java/graphql/execution/ExecutorServiceExecutionStrategy.java

```diff
@@ -58,8 +72,9 @@ public ExecutorServiceExecutionStrategy(ExecutorService executorService, DataFet
             ExecutionStrategyParameters newParameters = parameters
                     .transform(builder -> builder.field(currentField).path(fieldPath));
 
-            Callable<ExecutionResult> resolveField = () -> resolveField(executionContext, newParameters).join();
-            futures.put(fieldName, executorService.submit(resolveField));
+            Callable<ExecutionResult> fieldResolver = () -> resolveField(executionContext, newParameters).join();
+            Callable<ExecutionResult> wrappedFieldResolver = callableWrapper.wrapCallable(fieldResolver);
+            futures.put(fieldName, executorService.submit(wrappedFieldResolver));
         }
```

Comment:
I understand your concern here.  That is you want to get specific logic in place inside the "Callables" that get run.

But can you do all this via having your own custom ExecutorService?

Here its calls  `callableWrapper.wrapCallable(fieldResolverCallable);`

why can you do that INSIDE a custom executor service you wire in?

     customExecutorService.submit(fieldResolverCallable)

The result would be the same no?